ncdfchecker.py

```diff
@@ -189,16 +193,66 @@ def check_globals(product, constraints, skip=["short_name"], strict=False,
     return errcount, warncount
 
 
-def check_stepsize(data, stepsize):
+def get_month_midpoints(monthslist):
+    """Return a list of dates that correspond to the middle of the month."""
+    def midpoint(start, end): return start + (end - start) / 2
+
+    midmonths = [midpoint(monthslist[i], monthslist[i+1])
+                 for i in range(len(monthslist)-1)]
+
+    return midmonths
+
+
+def get_expected_monthly_stepsize(initdate, runlength):
     """
-    Helper routine to check that the stepsize in some given data is equal to
-    a given value
+    Using the model initialisation date and run length, calculate the
+    expected step size between entries in a monthly file. Use the midpoint
+    in each month to calculate said step size.
+
+    :arg: string initdate : Model initialisation date
+    :arg: int runlength   : Model run length in days
+
+    """
+    SECS_IN_HR = 60 * 60
+
+    initdate = dateutil.parser.parse(initdate)
```

Comment:
Use first date entry from the data provided for iteration.

Reply:
The forecast_reference_time is now used.